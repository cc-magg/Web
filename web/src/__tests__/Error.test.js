import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ErrorContainer from '../components/error/container/handle-error'
import ErrorComponent from '../components/error/component/regular-error'

test('Error should render as expected', () => {
    const wrapper = shallow(<ErrorContainer />);
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
});

test('ErrorContainer should render as expected', () => {
    const wrapper = shallow(<ErrorComponent errorInfo={{ "componentStack": 'error testing text' }} error={'ReferenceError: HomeComponnt is not defined'} />);
    const textContainer = wrapper.find('.errorLocation');
    const tree = toJson(wrapper);

    expect(textContainer.length).toBe(1);
    expect(textContainer.render().text()).toEqual('Here is the error: error testing text');
    expect(tree).toMatchSnapshot();
});
