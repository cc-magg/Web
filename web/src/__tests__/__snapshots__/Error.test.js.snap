// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render as expected 1`] = `<div />`;

exports[`ErrorContainer should render as expected 1`] = `
<div
  className="jsx-1329592948"
>
  <h3
    className="jsx-1329592948 header"
  >
    ERROR
  </h3>
  <div
    className="jsx-1329592948 errorContainer"
  >
    <div
      className="jsx-1329592948 errorCard"
    >
      <p
        className="jsx-1329592948"
      >
        ReferenceError: HomeComponnt is not defined
      </p>
      <div
        className="jsx-1329592948 errorLocation"
      >
        Here is the error: 
        error testing text
      </div>
    </div>
  </div>
  <JSXStyle
    id="311577014"
  >
    .header.jsx-1329592948{width:100%;padding:10px;background-color:#455498;color:#fff;text-align:center;}.errorContainer.jsx-1329592948{width:100%;margin-top:50px;}.errorCard.jsx-1329592948{max-width:500px;margin:0 auto;padding:20px;background-color:#fff;border-radius:5px;box-shadow:0 4px 8px 0 rgba(0,0,0,0.2);}.errorLocation.jsx-1329592948{border-radius:5px;padding:10px;background-color:#feb5b9;}
  </JSXStyle>
  <JSXStyle
    id="2727828427"
  >
    body{background-color:#87c2e3;}
  </JSXStyle>
</div>
`;
